### PR TITLE
Support for Siemens IOT2000

### PR DESCRIPTION
**Pull request by telphan**
Connects to resin-os/resin-os-device-support#60

### CODE REVIEW

**Comment by telphan** [comment]
@resin-jenkins retest this please

**Comment by telphan** [comment]
@resin-jenkins retest this please

**Comment by telphan** [comment]
@resin-jenkins retest this please

**Comment by telphan** [comment]
@resin-jenkins retest this please

File: layers/meta-resin-iot2000/recipes-core/images/resin-image-initramfs.bbappend
Comment (floion) [inline layers/meta-resin-iot2000/recipes-core/images/resin-image-initramfs.bbappend]: Typos in commit log

File: VERSION
Comment (floion) [inline VERSION]: Please change to meta-resin v2.7.4 and use this CHANGELOG style: https://github.com/resin-os/resin-raspberrypi/blob/master/CHANGELOG.md

File: iot2000.coffee
Comment (floion) [inline iot2000.coffee]: Please change the getting started links to something like: https://github.com/resin-os/resin-raspberrypi/blob/master/raspberrypi.coffee#L21

Coordinate with @zwalchuk about these links

File: iot2000.coffee
Comment (floion) [inline iot2000.coffee:23]: Any restriction for morty? Can we switch to pyro? meta-intel already supports pyro

**Review by floion — CHANGES_REQUESTED** [review]

File: VERSION
Comment (floion) [inline VERSION]: Please change to meta-resin v2.7.4 and use this CHANGELOG style: https://github.com/resin-os/resin-raspberrypi/blob/master/CHANGELOG.md

**Review by floion — CHANGES_REQUESTED** [review]

File: layers/meta-resin-iot2000/recipes-core/images/resin-image-initramfs.bbappend
Comment (floion) [inline layers/meta-resin-iot2000/recipes-core/images/resin-image-initramfs.bbappend]: Typos in commit log

File: iot2000.coffee
Comment (floion) [inline iot2000.coffee]: Please change the getting started links to something like: https://github.com/resin-os/resin-raspberrypi/blob/master/raspberrypi.coffee#L21

Coordinate with @zwalchuk about these links

**Review by floion — CHANGES_REQUESTED** [review]

File: iot2000.coffee
Comment (telphan) [reply iot2000.coffee:23]: meta-iot2000 only supports morty

File: layers/meta-resin-iot2000/conf/samples/bblayers.conf.sample
Comment (floion) [inline layers/meta-resin-iot2000/conf/samples/bblayers.conf.sample:20]: Please also include meta-rust. Like this board does: https://github.com/resin-os/resin-artik/blob/master/layers/meta-resin-artik/conf/samples/bblayers.conf.sample#L19

File: layers/meta-resin-iot2000/conf/samples/bblayers.conf.sample
Comment (floion) [inline layers/meta-resin-iot2000/conf/samples/bblayers.conf.sample]: The above should be replaced by:

POKY_BBLAYERS_CONF_VERSION = "2"

File: layers/meta-resin-iot2000/conf/samples/bblayers.conf.sample
Comment (floion) [inline layers/meta-resin-iot2000/conf/samples/bblayers.conf.sample]: This should instead be "meta-poky"

File: layers/meta-resin-iot2000/conf/samples/bblayers.conf.sample
Comment (floion) [inline layers/meta-resin-iot2000/conf/samples/bblayers.conf.sample]: This should instead be "meta-poky"

File: layers/meta-resin-iot2000/conf/samples/local.conf.sample
Comment (floion) [inline layers/meta-resin-iot2000/conf/samples/local.conf.sample]: The above 2 lines should be removed as meta-resin now uses package_ipk

**Review by floion — CHANGES_REQUESTED** [review]

File: .gitmodules
Comment (floion) [inline .gitmodules]: This commit should be part of the other commit which adds the gitmodules: https://github.com/resin-os/resin-iot2000/pull/1/commits/0be5e793ceda3f5f24353bcca53cde167fd7fa8f

Please merge the 2

**Review by floion — CHANGES_REQUESTED** [review]

File: layers/meta-resin-iot2000/recipes-core/images/resin-image.bbappend
Comment (floion) [inline layers/meta-resin-iot2000/recipes-core/images/resin-image.bbappend]: This should only append "resinos-img". Why the need to explicitly also append "hostapp-ext4" ?

**Review by floion — CHANGES_REQUESTED** [review]

File: layers/meta-resin-iot2000/recipes-core/images/resin-image-initramfs.bbappend
Comment (floion) [inline layers/meta-resin-iot2000/recipes-core/images/resin-image-initramfs.bbappend:1]: Typos in commit log

**Review by floion — CHANGES_REQUESTED** [review]

File: layers/meta-resin-iot2000/recipes-support/hostapp-update-hooks/files/99-resin-grub
Comment (floion) [inline layers/meta-resin-iot2000/recipes-support/hostapp-update-hooks/files/99-resin-grub]: Please also submit a PR with this file in meta-resin as there are more boards that will use the grub hook so we don't duplicate code. Post the PR link here and until meta-resin merges your PR, we will use this hook here too.

**Review by floion — CHANGES_REQUESTED** [review]

File: iot2000.coffee
Comment (floion) [inline iot2000.coffee:17]: Did you also ping Zach about these?

File: iot2000.coffee
Comment (floion) [inline iot2000.coffee]: I believe the state is "experimental" ?

File: CHANGELOG.md
Comment (floion) [inline CHANGELOG.md]: Wrong commit log

**Review by floion — CHANGES_REQUESTED** [review]

**Comment by floion** [comment]
@telphan I think you also need to submit a PR in meta-resin for adding the new supervisor build for this board, here:

https://github.com/resin-os/meta-resin/blob/master/meta-resin-common/recipes-containers/docker-disk/docker-resin-supervisor-disk.bb#L10 and the name I believe is resin/iot2000-supervisor , ( @nghiant2710 please confirm )

File: iot2000.coffee
Comment (zwalchuk) [reply iot2000.coffee:17]: @telphan I need info for two sections to create a getting started guide: What You Will Need, and Setting Up Your Device. Feel free to copy the sections from the devices that are closest to this new device and make modifications as necessary

File: CHANGELOG.md
Comment (floion) [reply CHANGELOG.md]: git commit log still says v2.7.2+rev1
Please pay attention, previous comment said this already

**Review by floion — CHANGES_REQUESTED** [review]

File: iot2000.coffee
Comment (floion) [reply iot2000.coffee]: This still shows the wrong state. Are you sure the comments have been addressed @telphan ?

File: iot2000.coffee
Comment (floion) [reply iot2000.coffee:23]: Tried building it with pyro and had compile issues?

File: iot2000.coffee
Comment (agherzan) [reply iot2000.coffee]: Ping @telphan

File: iot2000.coffee
Comment (agherzan) [inline iot2000.coffee]: What are these aliases?

File: iot2000.coffee
Comment (telphan) [reply iot2000.coffee]: @floion @agherzan I am waiting on the supervisor to merged, the changes are made, just not pushed.

File: iot2000.coffee
Comment (telphan) [reply iot2000.coffee]: I have this fixed, locally.

File: iot2000.coffee
Comment (agherzan) [reply iot2000.coffee:23]: I don't see any requirement in https://github.com/siemens/meta-iot2000 about build system version? Could you please detail, @telphan ?

File: .gitmodules
Comment (agherzan) [inline .gitmodules:10]: It would be nice to use pyro so we can get rid of this submodule.

File: layers/meta-resin-iot2000/conf/samples/bblayers.conf.sample
Comment (agherzan) [reply layers/meta-resin-iot2000/conf/samples/bblayers.conf.sample]: Please use an autogenerated file from a build.

File: .gitmodules
Comment (telphan) [reply .gitmodules:10]: The meta-iot2000 only supports morty

File: iot2000.coffee
Comment (floion) [reply iot2000.coffee:23]: So did it compile with pyro? I know you were trying that a few days ago

Comment (telphan) [reply iot2000.coffee:23]: I am facing issues with that build. The pyro update will follow after this initial release.

File: iot2000.coffee
Comment (floion) [reply iot2000.coffee:17]: @telphan did you provide to @zwalchuk what he has asked for?

File: .gitmodules
Comment (floion) [inline .gitmodules:9]: Please use HEAD of master for resin-yocto-scripts

**Review by floion — CHANGES_REQUESTED** [review]

File: iot2000.coffee
Comment (telphan) [reply iot2000.coffee:17]: @floion provided internally.

File: .gitmodules
Comment (agherzan) [reply .gitmodules:10]: I don't think that it's true: https://github.com/siemens/meta-iot2000/tree/master/meta-iot2000-bsp . And it seems to be confirmed that it should work with pyro too in https://github.com/siemens/meta-iot2000/issues/60 .

File: layers/meta-resin-iot2000/recipes-support/hostapp-update-hooks/hostapp-update-hooks.bbappend
Comment (floion) [inline layers/meta-resin-iot2000/recipes-support/hostapp-update-hooks/hostapp-update-hooks.bbappend]: This needs to go away as we have the hooks in meta-resin

File: layers/meta-resin-iot2000/recipes-support/hostapp-update-hooks/files/99-resin-grub
Comment (floion) [reply layers/meta-resin-iot2000/recipes-support/hostapp-update-hooks/files/99-resin-grub]: This entire file needs to go away now because we have this hook in meta-resin

**Review by floion — CHANGES_REQUESTED** [review]

File: CHANGELOG.md
Comment (floion) [reply CHANGELOG.md]: This should say 2.7.6

**Review by floion — CHANGES_REQUESTED** [review]

**Review by floion — APPROVED** [review]

**Comment by floion** [comment]
@resin-jenkins retest this please

**Comment by lfarkas** [comment]
which version (will) include this support? we need iot202 actually: http://w3.siemens.com/mcms/sce/en/simatic-iot2020/Pages/default.aspx but i hope they are the same base os.

**Comment by floion** [comment]
@lfarkas this should work with 2020 and 2040

**Pull request merged**
